### PR TITLE
CR-15777

### PR DESCRIPTION
**Pull request by kim-codefresh**
Signed-off-by: kim-codefresh <kim.aharfi@codefresh.io>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,4 +1,4 @@
-VERSION=v0.4.9
```
Comment (roi-codefresh) [inline Makefile:1]: You don't need to bump the version, it is done when releasing a new version. Please remove bump

File: pkg/git/repository.go [hunk 2]
```diff
@@ -244,6 +244,13 @@ func (o *CloneOptions) GetRepo(ctx context.Context) (Repository, fs.FS, error) {
 		}
 	}
 
+	if o.CloneForWrite {
```
Comment (roi-codefresh) [inline pkg/git/repository.go:247]: If we created the repo in line 230 there is no point in validating write permission. Please take it into consideration.

File: pkg/git/repository.go [hunk 3]
```diff
@@ -244,6 +244,13 @@ func (o *CloneOptions) GetRepo(ctx context.Context) (Repository, fs.FS, error) {
...
+		err = validateRepoWritePermission(ctx, r)
+		if err != nil {
+			return nil, nil, fmt.Errorf("failed to validate repository write permissions: %w", err)
```
Comment (roi-codefresh) [inline pkg/git/repository.go:250]:
```suggestion
			return nil, nil, fmt.Errorf("failed to validate repository write permission: %w", err)
```

File: pkg/git/repository.go [hunk 4]
```diff
@@ -252,6 +259,49 @@ func (o *CloneOptions) GetRepo(ctx context.Context) (Repository, fs.FS, error) {
...
+		return fmt.Errorf("failed getting repository certificates: %w", err)
+	}
+
+	err = r.Repository.PushContext(ctx, &gg.PushOptions{
+		Auth:     getAuth(r.auth),
+		Progress: r.progress,
+		CABundle: cert,
+	})
+	if err != nil {
+		if !errors.Is(err, gg.NoErrAlreadyUpToDate) {
```
Comment (roi-codefresh) [inline pkg/git/repository.go:274]: why would there already be a commit to push? we just cloned it on line 221. why not go straight to `PushDummyCommit`

File: pkg/git/repository.go [hunk 5]
```diff
@@ -252,6 +259,49 @@ func (o *CloneOptions) GetRepo(ctx context.Context) (Repository, fs.FS, error) {
...
+			// means there was already a commit to push, and it failed to push it
+			return err
+		}
+		// no commit to push, all up to date
+		if err := r.PushDummyCommit("Validating repository write permissions", ctx); err != nil {
```
Comment (roi-codefresh) [inline pkg/git/repository.go:279]:
```suggestion
		if err := r.PushDummyCommit("Validating repository write permission", ctx); err != nil {
```

File: pkg/git/repository.go [hunk 6]
```diff
@@ -252,6 +259,49 @@ func (o *CloneOptions) GetRepo(ctx context.Context) (Repository, fs.FS, error) {
...
+		CommitMsg: commitMsg,
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed pushing commit to repository: %w", err)
+	}
+
+	if err != nil {
+		log.G().Warnf("failed to reset commit: %v", err)
+	}
```
Comment (roi-codefresh) [inline pkg/git/repository.go:299]: remove unnecessary check here

File: pkg/git/repository.go (hunk 4 above)
Comment (kim-codefresh) [reply pkg/git/repository.go:274]: yeah you actually right, so I guess it can come to almost one line

**Review by roi-codefresh — APPROVED** [review]

**Pull request merged**